tests/test_routes/test_motors_route.py

```diff
@@ -0,0 +1,326 @@
+from fastapi import HTTPException
+from lib.models.motor import Motor
+from lib.controllers.motor import MotorController
+from lib.views.motor import (
+    MotorKinds,
+    MotorCreated,
+    MotorUpdated,
+    MotorDeleted,
+    MotorSummary,
+    MotorView,
+)
+from lib import app
+
+client = TestClient(app)
+
+
+@pytest.fixture
+def stub_motor():
+    motor = Motor(
+        thrust_source=[[0, 0]],
+        burn_time=0,
+        nozzle_radius=0,
+        dry_mass=0,
+        dry_inertia=[0, 0, 0],
+        center_of_dry_mass_position=0,
+    )
+    motor_json = motor.model_dump_json()
+    return json.loads(motor_json)
+
+
+@pytest.fixture
+def stub_motor_summary():
+    motor_summary = MotorSummary()
+    motor_summary_json = motor_summary.model_dump_json()
+    return json.loads(motor_summary_json)
+
+
+def test_create_motor(stub_motor):
+    with patch.object(
+        MotorController,
+        "create_motor",
+        return_value=MotorCreated(motor_id="123"),
+    ) as mock_create_motor:
+        with patch.object(
+            Motor, "set_motor_kind", side_effect=None
+        ) as mock_set_motor_kind:
+            response = client.post(
+                "/motors/", json=stub_motor, params={"motor_kind": "HYBRID"}
+            )
+            assert response.status_code == 200
+            assert response.json() == {
+                "motor_id": "123",
+                "message": "Motor successfully created",
+            }
+            mock_set_motor_kind.assert_called_once_with(MotorKinds.HYBRID)
+            mock_create_motor.assert_called_once_with(Motor(**stub_motor))
+
+
+def test_create_motor_optional_params():
+    test_object = {
+        "thrust_source": [[0, 0]],
+        "burn_time": 1,
+        "nozzle_radius": 1,
+        "dry_mass": 1,
+        "center_of_dry_mass_position": 1,
+        "dry_inertia": [0, 0, 0],
+        "interpolation_method": "LINEAR",
+        "coordinate_system_orientation": "NOZZLE_TO_COMBUSTION_CHAMBER",
+        "reshape_thrust_curve": False,
+    }
+    with patch.object(
+        MotorController,
+        "create_motor",
+        return_value=MotorCreated(motor_id="123"),
+    ) as mock_create_motor:
+        with patch.object(
+            Motor, "set_motor_kind", side_effect=None
+        ) as mock_set_motor_kind:
+            response = client.post(
+                "/motors/", json=test_object, params={"motor_kind": "HYBRID"}
+            )
+            assert response.status_code == 200
+            assert response.json() == {
+                "motor_id": "123",
+                "message": "Motor successfully created",
+            }
+            mock_set_motor_kind.assert_called_once_with(MotorKinds.HYBRID)
+            mock_create_motor.assert_called_once_with(Motor(**test_object))
+
+
+def test_create_motor_invalid_input():
+    response = client.post(
+        "/motors/", json={"burn_time": "foo", "nozzle_radius": "bar"}
+    )
+    assert response.status_code == 422
+
```

Comment:
Not necessarily, this would mean testing the extension of pydantic validations which is not the goal. The goal is to assess if the API has the capability of validating the input schema.